### PR TITLE
migrate RecoPixelVertexing/PixelLowPtUtilities to esConsumes

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all. 
Part of #31061.
I migrated only the "easy" ones.  Classes inheriting from base classes needing modifications outside of this package will have to be dealt with at a later time.

#### PR validation:

Run standard limited matrix validation:
```
runTheMatrix.py -l limited -j 8 --ibeos
```

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, no backport needed

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelVertexProducerClusters.h [hunk 1]
```diff
@@ -1,11 +1,13 @@
 #ifndef PixelVertexProducerClusters_H
 #define PixelVertexProducerClusters_H
 
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
 #include "FWCore/Framework/interface/global/EDProducer.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
-
 #include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
 
+class TrackerGeometry;
```
Comment (jpata) [inline RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelVertexProducerClusters.h:9]: I think we don't allow forward declaring classes from other packages [1]. Was there a reason to not include the header?

[1] (https://cms-sw.github.io/cms_coding_rules.html 4.7)

File: RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h [hunk 2]
```diff
@@ -40,6 +42,12 @@ class StripSubClusterShapeFilterBase {
                    const GlobalVector &gdir,
                    bool mustProject = false) const;
 
+  // esConsumes tokens
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> m_topoToken;
+  edm::ESGetToken<ClusterShapeHitFilter, CkfComponentsRecord> m_csfToken;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> m_geomToken;
+  edm::ESGetToken<SiStripNoises, SiStripNoisesRcd> m_stripNoiseToken;
```
Comment (jpata) [inline RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h:49]:
```suggestion
  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoToken_;
  edm::ESGetToken<ClusterShapeHitFilter, CkfComponentsRecord> csfToken_;
  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
  edm::ESGetToken<SiStripNoises, SiStripNoisesRcd> stripNoiseToken_;
```
would this be acceptable here, to stick with the trailing underscore as a member identifier?

File: RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h [hunk 3]
```diff
@@ -9,6 +9,8 @@
 #include "FWCore/Utilities/interface/EDGetToken.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "RecoTracker/TkSeedingLayers/interface/SeedComparitor.h"
+#include "CondFormats/SiStripObjects/interface/SiStripNoises.h"
+#include "CondFormats/DataRecord/interface/SiStripNoisesRcd.h"
```
Comment (jpata) [inline RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h:13]: I think these two should packages be added also to https://github.com/cms-sw/cmssw/blob/master/RecoPixelVertexing/PixelLowPtUtilities/BuildFile.xml (to be explicit in package dependencies and simplify later cleanups)

**Review by jpata — COMMENTED** [review]
Thanks a lot for this migration! I don't see any problem in the reco tests.

File: RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelVertexProducerClusters.h (hunk 1 above)
Comment (mmusich) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelVertexProducerClusters.h]: done, though there are other examples of that in this package:

https://github.com/cms-sw/cmssw/blob/dfffa4a11f400b84082bfd53a6de14ced9800d59/RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelVertexProducerClusters.h#L15

File: RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h (hunk 2 above)
Comment (mmusich) [reply RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h]: done.

File: RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h (hunk 3 above)
Comment (mmusich) [reply RecoPixelVertexing/PixelLowPtUtilities/interface/StripSubClusterShapeTrajectoryFilter.h:13]: done, as well as re-ordered alphabetically the `BuildFile.xml`

**Comment by VinInn** [comment]
> On 15 Mar, 2021, at 12:53 PM, Joosep Pata ***@***.***> wrote:
> 
> In RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelVertexProducerClusters.h:
> 
> >  #include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
>  
> +class TrackerGeometry;
> 
> I think we don't allow forward declaring classes from other packages [1]. Was there a reason to not include the header?
> 
> [1] (https://cms-sw.github.io/cms_coding_rules.html 4.7)
> 
> 
May I ask what's the rationale behind such a rule?

v.

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
> May I ask what's the rationale behind such a rule? v.

perhaps @slava77 can clarify the reasoning behind this rule?

**Comment by slava77** [comment]
> > May I ask what's the rationale behind such a rule? v.
> 
> perhaps @slava77 can clarify the reasoning behind this rule?

please see https://github.com/cms-sw/cms-sw.github.io/issues/94#issuecomment-678368336
and related discussion in the thread.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jpata** [comment]
+reconstruction
- technical, migrates a number of classes in RecoPixelVertexing/PixelLowPtUtilities to esConsumes
- no reco changes expected/observed

Thanks a lot for this PR!

**Comment by qliphy** [comment]
+1

**Pull request merged**